### DISCUSSION TITLE
Clang-Tidy results mismatch

### DISCUSSION THREAD

**Discussion by H-G-Hristov** [microsoft/vscode-cpptools#9964] [Q&A]
On a file of mine I run the command:

clang-tidy -config-file=.clang-tidy '<path_to_file>/LaunchdPlist.cpp'

The output from this command is ver different than the Clang-Tidy warnings I see in the code editor. 
Also a lot of these warnings are false positives.

How should I interpret that situation? 

![Screenshot 2022-10-06 at 16 00 18](https://user-images.githubusercontent.com/47526411/194319198-4537fad8-88f0-4121-b3be-ece305801831.png)

BTW. Should I report bugs to Clang-Tidy or should I just disable/ignore some of these warnings/checks?

**Comment by sean-mcmanus** [comment]
You should only report a clang-tidy bug to https://github.com/llvm/llvm-project/issues if you have a repro using clang-tidy on the command line.

Our extension does some different configuration and processing of the output on top of clang-tidy, so the difference might either be a bug with our handling or "by design" for some reason. If you set C_Cpp.loggingLevel to Debug, the "C/C++" logging will show the command line we end up running and the raw output, which we may filter using our exclude settings, etc.

Can you provide more info on what the difference is?

**Reply by sean-mcmanus** [reply]
Oh, I just noticed the issue in your screenshot -- it looks like clang-tidy is reporting a variable as uninitialized when it's initialized -- that looks like a bug with clang-tidy itself, right?

**Reply by H-G-Hristov** [reply]
Yes, there I have a lot of these in this specific file but this one doesn't appear in the code editor. There is no ~~~~~~

**Comment by H-G-Hristov** [comment]
I created a LLVM issue here:

https://github.com/llvm/llvm-project/issues/58220

This is the console log from VSCode:

[clang-tidy-vscode.txt.zip](https://github.com/microsoft/vscode-cpptools/files/9732369/clang-tidy-vscode.txt.zip)

I can't recognise the clang-tidy command in the output.

**Reply by sean-mcmanus** [reply]
The clang-tidy command line can be found by searching for "clang-tidy" in the logging, but the arguments are separated by line breaks.

**Reply by H-G-Hristov** [reply]
I supposed this is the command line generated by `cpptools`:

> /Users/hhhhh3/.vscode/extensions/ms-vscode.cpptools-1.13.2-darwin-arm64/bin/../LLVM/bin/clang-tidy
--export-fixes=/private/tmp/loose/cpptools/10908265598756619264/fixes6149156864.yaml
--quiet
/Users/hhhhh3/Projects/GitHub/nmylibmylib/src/utilities/src/macos/LaunchdPlist.cpp

I invoke Clang-Tidy with the command below. My root (and only) `.clang-tidy` file is is the root of the `CMakeLists.txt` project:

> hhhhh@Hristos-MacBook-Pro mylib % clang-tidy -config-file=.clang-tidy '/Users/hhhhh/Projects/GitHub/mylib/src/utilities/src/macos/LaunchdPlist.cpp'

I suppose that catching what fixesXXXXXXX.yaml contains will be difficult so I'm going to ask instead: What does fixesXXXXX.yaml contain how is that any different than my own `.clang-tidy`? I'd expect that what cpptools does will deliver the same result as my own `.clang-tidy` specifies.

What I observe in reality is that there are some errors reported when I invoke the above command which are not displayed in the code editor.
Most of the error reported for the file I test are cppcoreguidelines-init-variables false positives IMO, yet only some are displayed by cpptools in the code editor. What would be the reason for that - e.g. post processing?

**Reply by sean-mcmanus** [reply]
We use "--export-fixes" for the clang-tidy fixes for code actions. In generally, it shouldn't cause any change in results, unless there's some bug. You could try using that in your command line as well if you wanted.

Do you see the "missing errors" in the C/C++ logging output with C_Cpp.logigngLevel set to "Debug"? The logging output should contain the raw clang-tidy output. It should filter out results from any files.exclude, C_Cpp.files.exclude, or C_Cpp.codeAnalysis.exclude.
